MSAL/src/native_auth/public/state_machine/state/MFAStates.swift

```diff
@@ -24,51 +24,97 @@
 
 import Foundation
 
+/// Base class for MFA state
+@objcMembers public class MFABaseState: MSALNativeAuthBaseState {
+    let controller: MSALNativeAuthMFAControlling
+    let scopes: [String]
+
+    init(
+        controller: MSALNativeAuthMFAControlling,
+        scopes: [String],
+        continuationToken: String,
+        correlationId: UUID) {
+        self.controller = controller
+        self.scopes = scopes
+        super.init(continuationToken: continuationToken, correlationId: correlationId)
+    }
+
+    func baseRequestChallenge(authMethod: MSALAuthMethod?, delegate: MFARequestChallengeDelegate) {
```

Comment:
This is not public right?

Reply:
Correct, can be called only from other states